### PR TITLE
don't log repeatedly for task ext activation

### LINKED ISSUE #229155: `Activating task providers npm` shows up twice?
Repro:

1. Reload VS Code window, the tasks output channel has a single `Activating task providers npm`
2. Ctrl+shift+b, 🐛 it shows up again

Image

This might just be logging too aggressively or we might be doing unneeded work here? If task providers are already activated we should not lot the second time, if something extra is happening on the second time we should make the logs more descriptive.

### PR DESCRIPTION
**Pull request by meganrogge**
fix #229155

### CODE REVIEW

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 1]
```diff
@@ -616,19 +616,18 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
...
 
 	private async _activateTaskProviders(type: string | undefined): Promise<void> {
-		if (type && this._activatedTaskProviders.has(type)) {
-			return;
-		}
 		// We need to first wait for extensions to be registered because we might read
 		// the `TaskDefinitionRegistry` in case `type` is `undefined`
 		await this._extensionService.whenInstalledExtensionsRegistered();
-		this._log('Activating task providers ' + (type ?? 'all'));
+		if (!type || !this._activatedTaskProviders.has(type)) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:622]:
```suggestion
		if (!!type && !this._activatedTaskProviders.has(type)) {
```

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 2]
```diff
@@ -616,19 +616,18 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
...
 
 	private async _activateTaskProviders(type: string | undefined): Promise<void> {
-		if (type && this._activatedTaskProviders.has(type)) {
-			return;
-		}
 		// We need to first wait for extensions to be registered because we might read
 		// the `TaskDefinitionRegistry` in case `type` is `undefined`
 		await this._extensionService.whenInstalledExtensionsRegistered();
-		this._log('Activating task providers ' + (type ?? 'all'));
+		if (!!type && !this._activatedTaskProviders.has(type)) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:622]:
```suggestion
		if (type && !this._activatedTaskProviders.has(type)) {
```

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 3]
```diff
@@ -616,19 +616,18 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
...
+		if (type && !this._activatedTaskProviders.has(type)) {
+			this._log('Activating task providers ' + (type ?? 'all'));
+		}
 		const result = await raceTimeout(
 			Promise.all(this._getActivationEvents(type).map(activationEvent => this._extensionService.activateByEvent(activationEvent))),
 			5000,
 			() => console.warn('Timed out activating extensions for task providers')
 		);
-		if (result && type) {
+		if (type && result && !this._activatedTaskProviders.has(type)) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:630]: this `this._activatedTaskProviders.has` check doesn't really matter but can't hurt

File: src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts [hunk 4]
```diff
@@ -616,20 +616,20 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
...
 	private async _activateTaskProviders(type: string | undefined): Promise<void> {
-		if (type && this._activatedTaskProviders.has(type)) {
-			return;
-		}
 		// We need to first wait for extensions to be registered because we might read
 		// the `TaskDefinitionRegistry` in case `type` is `undefined`
 		await this._extensionService.whenInstalledExtensionsRegistered();
-		this._log('Activating task providers ' + (type ?? 'all'));
+		const hasLoggedTypeActivation = type && !this._activatedTaskProviders.has(type);
+		if (hasLoggedTypeActivation) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts:623]:
```suggestion
		if (!hasLoggedTypeActivation) {
```

**Review by joshspicer — APPROVED** [review]

**Pull request merged**